File: 2024/Jan/day09.py
class Solution:
    def search(self, pat, txt):
        # code here
        ind = []
        res = []
        for i in range(len(txt)):
            if txt[i]==pat[0]:
                ind.append(i)
        for i in ind:
            if i+len(pat)<=len(txt) and txt[i:i+len(pat)]== pat:
                res.append(i+1)
        return res

File: 2024/March/day09.cpp
class Solution{
  public:
    char nthCharacter(string s, int r, int n) {
        //code here
        if(r==0 || n==0)
            return s[n];
        long long j=(1ll<<r);
        int i=n/j;
        n-=(i*j);
        if(s[i]=='1')
            return nthCharacter("10",r-1,n);
        else
            return nthCharacter("01",r-1,n);
        return '2';
    }
};
